archive/path.go

```diff
@@ -0,0 +1,123 @@
+package archive
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	errTooManyLinks = errors.New("too many links")
+)
+
+// rootPath joins a path with a root, evaluating and bounding any
```

Comment:
I assumed he meant `bounding` as in a bounding box that won't allow links outside of root